src/validate.ts

```diff
@@ -1,48 +1,76 @@
 import * as React from 'react';
 import { get, isEmpty, forEach } from 'lodash';
 import { hasError } from './validators';
-import { ComponentInstance } from './interfaces';
+import { ComponentInstance, ValidationType, ValidationRuleResult, IFormProps } from './interfaces';
 
 const hoistNonReactStatics = require('hoist-non-react-statics');
 
-const getValidationErrors = (rules: any[], model: { [name in string]: any }, allProps: any) =>
-    rules.reduce((errors, [rule, field]) => {
-        const isValid = rule(model, allProps);
+const getValidationResult = ({
+    validationType,
+    rules,
+    model,
+    allProps
+}: {
+    validationType: ValidationType;
+    rules: ValidationRuleResult[];
+    model: { [name in string]: any };
+    allProps: any;
+}) =>
+    rules
+        .filter(([rule, field, type = ValidationType.Error]) => type === validationType)
```

Comment:
This filter is new. Also defaulting `type` to Error which doesn't feel great for some reason.